src/main/java/org/sonar/plugins/github/GitHubPluginConfiguration.java

```diff
@@ -117,4 +121,59 @@ public boolean tryReportIssuesInline() {
     return !settings.getBoolean(GitHubPlugin.GITHUB_DISABLE_INLINE_COMMENTS);
   }
 
+    /**
+     * Checks if a proxy was passed with command line parameters or configured in the system.
+     * If only an HTTP proxy was configured then it's properties are copied to the HTTPS proxy (like SonarQube configuration)
+     * @return True iff a proxy was configured to be used in the plugin.
+     */
+  public boolean isProxyConnectionEnabled() {
+    if (System.getProperty("http.proxyHost") != null || System.getProperty("https.proxyHost") != null ||
+            System.getProperty("socksProxyHost") != null)
+    {
+        return true;
+    }
+    return false;
+  }
+
+
+  public Proxy getHttpProxy() {
+    try
+    {
+        if(System.getProperty("http.proxyHost") != null && System.getProperty("https.proxyHost") == null)
+        {
+            System.setProperty("https.proxyHost", System.getProperty("http.proxyHost"));
+        }
+
+        String proxyUser;
+        String proxyPass;
+
+        if(((proxyUser = System.getProperty("http.proxyUser")) != null && (proxyPass = System.getProperty("http.proxyPassword")) != null))
+        {
+            Authenticator.setDefault(
+                    new Authenticator() {
+                        @Override
+                        public PasswordAuthentication getPasswordAuthentication() {
+                            return new PasswordAuthentication(
+                                    proxyUser, proxyPass.toCharArray());
+                        }
+                    }
+            );
+        }
+        Proxy selectedProxy = ProxySelector.getDefault().select(new URI(endpoint())).get(0);
```

Comment:
Does select always return a collection with at least 1 element? Shouldn't you test first if the returned collection is empty?